### PR TITLE
uniform make file experience

### PR DESCRIPTION
**Pull request by wangxinyi7**
Make the makefile experience uniform
Changes proposed in this PR:
- introduce help target to display help info for each of targets
- tag the docker image

Ticket: https://hashicorp.atlassian.net/browse/NET-4834

Checklist:
- [ ] ~Tests added~
- [ ] [~CHANGELOG entry added~](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by curtbushko — CHANGES_REQUESTED** [review]
Looks pretty good.  Just a small request.

File: Makefile [hunk 1]
```diff
@@ -6,7 +6,7 @@ KIND_VERSION= $(shell ./control-plane/build-support/scripts/read-yaml-config.sh
 KIND_NODE_IMAGE= $(shell ./control-plane/build-support/scripts/read-yaml-config.sh acceptance/ci-inputs/kind-inputs.yaml .kindNodeImage)
 KUBECTL_VERSION= $(shell ./control-plane/build-support/scripts/read-yaml-config.sh acceptance/ci-inputs/kind-inputs.yaml .kubectlVersion)
 
-# ===========> Helm Targets
+##@ Helm Targets
```
[the remark this answers is not in the document]
Comment (wangxinyi7) [reply Makefile:10]: Yup, updated @curtbushko

**Review by curtbushko — APPROVED** [review]
Thanks!

**Comment by curtbushko** [comment]
@wangxinyi7 You'll want to backport these changes to all the release/1.1.x and up branches

**Comment by wangxinyi7** [comment]
@curtbushko Thanks for approving, just double check for the backport, I see there are backport labels of `1.1.x` `1.2.x` `1.3.x` in the dropdown menu, I need to select them and the following steps (open PR, approval and merging) will all be kicked off automatically right?

**Comment by curtbushko** [comment]
@wangxinyi7 Yes, those are the versions and it used the same backport assistant that the Consul repo uses. I recommend squashing all of our commits down to 1 commit before merging because the backport assistant has troubles sometimes and I find that it helps it get less confused...

**Comment by wangxinyi7** [comment]
Thanks for your info! @curtbushko

**Pull request merged**